opensearch/src/main/java/org/opensearch/sql/opensearch/data/value/OpenSearchExprValueFactory.java

```diff
@@ -252,7 +249,10 @@ private ExprValue parseTimestampString(String value, OpenSearchDateType dateType
    * @return time without timezone
    */
   private ExprValue parseTimeString(String value, OpenSearchDateType dateType) {
-    for (DateFormatter formatter : dateType.getAllNamedFormatters()) {
+    List<DateFormatter> formatters = dateType.getAllNamedFormatters();
+    formatters.addAll(dateType.getAllCustomFormatters());
+
+    for (DateFormatter formatter : formatters) {
```

Comment:
This loop repeated in 3 places - it is possible to move it to a new function to avoid code duplicates?

Reply:
Out of scope of the spike and will be part of the full implementation PR. Thanks for the comment.

Reply:
Fixed in be8c34f6f